### DISCUSSION TITLE
Query re: choosing background peaks for motif analysis between groups

### DISCUSSION THREAD

**Discussion by jarchana09** [stuart-lab/signac#1157] [Q&A] [answered]
Hi,
I am analyzing a 10X multiome (snRNA+ATAC) dataset with 2 groups (Stimulated and Unstimulated). I am not sure whether I am choosing the background correctly when analyzing motifs for each celltype. I have tried to subset the data and select background for each celltype. Please could you let me know if this approach is correct?
Please find included below the code I have used.
I would greatly appreciate any inputs.
Thank you so much for all your help and guidance.
 

```
#Motif analysis
enriched_motifs_list <- list()

for(i in unique(all_integrated_rnm$celltype){
  #subset results from running FindMarkers between groups (Stimulated vs. Unstimulated) for each cell type
  subset_top_da_peaks <- subset(top_da_peaks, cell_type %in% i)

  #set rownames of subset_top_da_peaks to peaks column values
  row.names(subset_top_da_peaks) <- subset_top_da_peaks$peaks
  
  #subset seurat integrated object to cell type
  subset_all_integrated_rnm <- subset(all_integrated_rnm, celltype %in% i)
  
  Idents(subset_all_integrated_rnm) <- "cluster1.1.group" #set identity to the metadata column of cell type + group id
  
   # find peaks open in Stimulated or Unstimulated cells
  open.peaks <- AccessiblePeaks(subset_all_integrated_rnm, idents = c(paste0(i, "_Stimulated"), 
                                                  paste0(i, "_Unstimulated")))
  
  # match the overall GC content in the peak set #https://github.com/timoast/signac/issues/151#issuecomment-624266123
  meta.feature <- GetAssayData(subset_all_integrated_rnm, assay = "peaks", slot = "meta.features") 
  peaks.matched <- MatchRegionStats(
  meta.feature = meta.feature[open.peaks, ],
  query.feature = meta.feature[row.names(subset_top_da_peaks), ],
  n = 50000
)

# test enrichment
enriched.motifs <- FindMotifs(
  object = subset_all_integrated_rnm,
  background=peaks.matched,
  features = row.names(subset_top_da_peaks), p.adjust.method = "BH") 
#(Source code for using all the row.names of DE results instead of only the top differentially accesible peaks : 
#https://github.com/timoast/signac/issues/1109#issuecomment-1121140510); #p.adjust.method = "BH" (default)

enriched.motifs$celltype <- i
enriched_motifs_list[[i]] <- enriched.motifs

}
```

**Comment by timoast** [comment, marked as answer]
Looks good to me

**Reply by jarchana09** [reply]
Thank you so much Tim. Really appreciate your inputs. I would be grateful if you could, i f possible, also kindly suggest whether the integration steps upstream of this analysis seem okay to you, as the pct values during differential expression analysis of the ATAC data were low. I had posted this question originally in the Seurat github page at: [https://github.com/satijalab/seurat/issues/6138](https://github.com/satijalab/seurat/issues/6138).

Thank you very much.

**Reply by timoast** [reply]
Seems ok to me. If you think the percent expressed values are not accurate I'd suggest looking at some of the peaks in a coverage plot

**Reply by jarchana09** [reply]
Sure Tim. Thank you very much for your guidance and insights.
Thank you.
